### DISCUSSION TITLE
在proxy函数里面直接调用被hook的函数不会死循环支持吗

### DISCUSSION THREAD

**Discussion by leinlin** [bytedance/android-inline-hook#23] [Q&A]
在proxy函数里面直接调用被hook的函数不会死循环支持吗

**Comment by caikelun** [comment]
unique 模式可以通过 hook 返回的 orig_addr 调用原函数，shared 模式可以通过 SHADOWHOOK_CALL_PREV 宏调用原函数，这样不会导致递归调用。
https://github.com/bytedance/android-inline-hook/blob/main/doc/manual.zh-CN.md

**Reply by leinlin** [reply]
大佬 你的orig_addr 我在调用完成函数后并不是 立刻有值的，我创建委托的时候会为空，有没有阻塞立刻获取的啊。

**Reply by leinlin** [reply]
我是给unity用的，在C#那边调用进来
![image](https://user-images.githubusercontent.com/30318270/201320248-8e30fc55-3b57-4822-954c-7101adc772d6.png)

**Reply by leinlin** [reply]
必须要在函数调用的里面写吗？

**Reply by leinlin** [reply]
![image](https://user-images.githubusercontent.com/30318270/201320845-44d8cba4-2509-4894-9a67-41d973f174b7.png)

**Reply by leinlin** [reply]
然后大佬 inline hook 模拟器上会闪退啊,原因是hook dlopen 进入的，然后就闪退了，不进入dlopen没有问题

**Comment by leinlin** [comment]
https://user-images.githubusercontent.com/30318270/201357401-32742d8c-8142-4f4d-a918-584cd540965b.mp4

**Reply by leinlin** [reply]
先hook symbol name，然后 dlopen 必定闪退啊

**Reply by leinlin** [reply]
原因好像是模拟器的dlopen是用 intel的x86_64架构做的汇编，而shadowhook没有判断出来，直接hook dlopen,一调用就寄

**Reply by caikelun** [reply]
shadowhook只支持arm/arm64，houdini环境的linker和大多数系统库都是x86的。
